### DISCUSSION TITLE
Generated types do not include relations

### DISCUSSION THREAD

**Discussion by isaacfink** [supabase/supabase#10371] [Questions]
Hey guys, I have a table `books` with a relation to `authors` I tried writing something like this:
```
async function test(){
        const {data, error} = await supabaseClient.from('books').select('*, author(*)')
        if(data){
            data.map(d => {
                console.log(d.author.first_name)
            })
        }
    }
```

but I get an error `property first_name does not exist on type number` what is the correct way to type this?

**Comment by isaacfink** [comment]
When I run it, the results are as expected, the authors are populated properly not just numbers, is this a bug with the types generation? I generated the types with the cli

**Comment by 5hee75** [comment]
I am also running into this issue - normally I can just ignore the type warning because I know the data is correct (although having the type checking would be nice).

It seems like the relation gets part of the type correct - if you inspect the inferred type coming back from the query it looks like it would almost be correct, but if you try to access the nested field it comes back as `any`.
